java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -477,7 +477,8 @@ public void visitLambdaExpression(LambdaExpressionTree tree) {
     LambdaExpressionTreeImpl lambdaExpressionTree = (LambdaExpressionTreeImpl) tree;
     if (lambdaExpressionTree.isTypeSet()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "visitLambdaExpression" is 19 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)